app/src/main/java/com/example/stromtracker/ui/raeume/raeumeErstellen/RaeumeErstellenFragment.kt

```diff
@@ -71,4 +89,23 @@ class RaeumeErstellenFragment : Fragment() {
 
         return root
     }
+
+    fun CustomTextListener(edit: EditText): EditText {
+        edit.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable) {
+                if (edit.text.toString().equals("Sonstiges")) {
+                    savebutton.visibility = View.INVISIBLE
+                    informationfield.text = raumnamesonstiges
+                } else {
+                    savebutton.visibility = View.VISIBLE
+                    informationfield.text = ""
+                }
+            }
+
+            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
+        })
+        return edit
+
+    }
```

Comment:
Das könnte ggf. ausgelagert werden (z.B. in ein Companion Objekt), da es genau die selbe Methode wie beim Bearbeiten ist.

Reply:
Inwiefern willst du das in ein companion object auslagern? Companion object ist ja nur die statische Bindung von Funktionen/Variablen an eine Klasse. Willst du dann eine extra Klasse mit statischen Funktionen machen?

Reply:
Ja genau. Weiß nicht ob man das in Kotlin auch anders lösen kann. Aber es wäre auf jeden Fall sinnvoll die identischen Funktionen auszulagern um Redundanz zu vermeiden. Ich habe das beim Urlaubsmodus beispielsweise mit einem Companion Object gelöst, was hier ja auch möglich wäre.

Reply:
Ja weiß nicht ob das Sinn macht. Aber wir können es ja auf die optional Liste setzen.